api/src/main/java/org/openmrs/LocationAttributeType.java

```diff
@@ -9,16 +9,35 @@
  */
 package org.openmrs;
 
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.annotations.Parameter;
 import org.openmrs.attribute.AttributeType;
 import org.openmrs.attribute.BaseAttributeType;
 
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
 /**
  * A user-defined extension to the {@link Location} class.
  * @see AttributeType
  * @since 1.9
  */
+@Entity
+@Table(name = "location_attribute_type")
 public class LocationAttributeType extends BaseAttributeType<Location> implements AttributeType<Location> {
-	
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "location_attribute_type_id_seq")
+	@GenericGenerator(
```

Comment:
Is there any reason why this is different from both `VisitAttributeType` and `ProviderAttributeType`?

Reply:
There is no distinction between these entities. However, in the XML mapping, the following configuration for the ID generator has been employed:

`
<generator class="native">
    <param name="sequence">location_attribute_type_location_attribute_type_id_seq</param>
</generator>
`

Therefore, I opted for GenerationType.SEQUENCE to align with the specified sequence configuration in the XML mapping.

Reply:
@dkayiwa

Reply:
@dkayiwa any update on this ?